core/src/main/scala/org/http4s/blaze/pipeline/Stages.scala

```diff
@@ -63,6 +63,7 @@ sealed trait Stage {
     */
   def inboundCommand(cmd: InboundCommand): Unit = cmd match {
     case Connected => stageStartup()
+    case Disconnected => logger.debug("Received Disconnected command")
```

Comment:
But timeout stage is going to send a disconnect outbound, and if we ignore the disconnected inbound, the tail won't actually shut down.  This isn't going to work, is it?

Reply:
* The `NIO1HeadStage` gets the `Disconnect` and shuts the connection down in a timely fashion.
* The `Http1ServerStage` gets the `Disconnected` notice in a timely fashion, but does nothing.  
  * This might be a good place to cancel any running effect, which only the stage knows.
  * Should the tail shut down on `Disconnected` if it wasn't shut down already?  But only the stage knows that, too.

Reply:
`stageShutdown` is idempotent.  It (probably) doesn't send outbound commands.  So why isn't `stageShutdown()` still the proper default response to `Disconnected`?